### DISCUSSION TITLE
lammps with multi GPUs

### DISCUSSION THREAD

**Discussion by cui-hq** [deepmodeling/deepmd-kit#3076] [Q&A] [answered]
The size of my system is too big for one GPU when running MD with lammps, so I want to use two gpus to perform MD, but I still met the error "RESOURCE_EXHAUSTED: failed to allocate memory", my command in "CUDA_VISIBLE_DEVICES=0,1 nohup mpirun -n 1 lmp -i input.lammps &", memory for gpu0 is 11205MiB while 134MiB for gpu1, I want to know how to run MD with two gpus to allocate the memory reasonable. thank you.

**Comment by njzjz** [comment]
To use two GPUs you need to use two MPI ranks.

**Reply by cui-hq** [reply]
I use two MPI ranks with "mpirun -n 2 lmp -i input.lammps", but it's invalid.
![1703928669022](https://github.com/deepmodeling/deepmd-kit/assets/118978478/6e06afcb-5b43-44b6-8689-c91b5476e46d)

**Reply by cui-hq** [reply]
this picture is the training progress, I use CUDA_VISIBLE_DEVICES=2,3 nohup mpirun -n 2 dp train input.json & Thanks your reply

**Reply by njzjz** [reply]
Have you installed horovod?

**Reply by cui-hq** [reply]
yes,I have installed.

**Reply by cui-hq** [reply]
![Uploading 1704161143716.png…]()

**Reply by njzjz** [reply, marked as answer]
Perhaps due to #2912?

**Reply by cui-hq** [reply]
thanks for your reply, in my system, openmpi was used.

**Comment by cui-hq** [comment]
![1704161143716](https://github.com/deepmodeling/deepmd-kit/assets/118978478/c25089cb-4c9e-439e-8d76-ee80711595e5)
